lib/io/gpio.cpp

```diff
@@ -2,20 +2,22 @@
 
 namespace hyped::io {
 
-Gpio::Gpio()
+Gpio::Gpio(hyped::utils::ILogger *log) : log_(log)
```

Comment:
I dont like that this is a pointer. The logger wont hold a lot of data, right? So passing it by value should be fine.